test/test_testrail_reporter.py

```diff
@@ -266,6 +266,31 @@ def test_config_file_without_projects(self):
         )
         self.assertEqual(exception_message, context.exception.args[0])
 
+    def test_config_file_with_project_as_object(self):
+        """
+        Ensure that an exception is created for an invalid config with projects
+        containing an `object` instead of an `array` of projects.
+        """
+        self.addCleanup(partial(os.remove, u"testrail.yml"))
+        data = {
+            "base_url": "https://test.testrail.net",
+            "projects": {
+                "name": "Test project",
+                "suite_id": 11,
+                "id": 1,
+                "allowed_branch_pattern": ".*",
+            },
+        }
```

Comment:
Ah, so this generates:

```
projects:
    name: 'Test project'
    id: 1
    suite_id: 11
    allowed_branch_pattern: '.*'
```

When we should have:
```
projects:
  -
    name: 'Test project'
    id: 1
    suite_id: 11
    allowed_branch_pattern: '.*'
```

Is that right? That's a bit subtle, I wonder if we can make that easier to write out somehow 🤔

Reply:
Pretty sure you can also write this like:

```
projects:
  - name: 'Test project'
     id: 1
     suite_id: 11
     allowed_branch_pattern: '.*'
```

Which might be easier to follow in the readme.